### PR TITLE
adding an Arabic vocab file

### PR DESCRIPTION
**Pull request by mzeidhassan**
I have added Arabic alphabet plus some Farsi alphabet because they may exist in many Arabic texts. I also added "Hindi" numbers and Arabic diacritics. Arabic uses connected shaping when it comes to characters, so characters are not isolated as in Latin-based languages. Hope this won't be an issue.  Please let me know if you have any questions.

### CODE REVIEW

**Review by charlesmindee — CHANGES_REQUESTED** [review]
Hi, thanks for the PR!

Could you please move this to `doctr/datasets/vocabs.py` ?

And as you mentioned, your are aggregating Hindi + Arabic + Farsi characters if I understood well, would it be possible to split those entries in the dictionary right there (below ancient greek):

```
VOCABS: Dict[str, str] = {
    'digits': string.digits,
    'ascii_letters': string.ascii_letters,
    'punctuation': string.punctuation,
    'currency': '£€¥¢฿',
    'ancient_greek': 'αβγδεζηθικλμνξοπρστυφχψωΑΒΓΔΕΖΗΘΙΚΛΜΝΞΟΠΡΣΤΥΦΧΨΩ',
}
```
For instance: 'hindi_letters', 'arabic_diacritics', etc... And try to be as exhaustive as possibe for each sub class. Then you can add below the dictionnary your full vocab:

`VOCABS['arabic'] = VOCABS['hindi_letters'] + VOCABS['farsi'] + VOCABS['arabic_diacritics'] + ...`

Thanks for that ! :pray:

**Comment by mzeidhassan** [comment]
Thanks @charlesmindee for your help and guidance. I appreciate it. I am new to PR pushing, so I hope I am doing it right.

Please find the changes I made here 
https://github.com/mindee/doctr/commit/3faad13ce1f449dce0b91f5da93013d5e18137b5

Instead of creating an new item "Arabic" numbers, I am using "VOCABS['digits']". 

Please let me know if there is anything else that I need to do at my end.

Thanks,
Mohamed

File: arabic_vocab [hunk 1]
```diff
@@ -0,0 +1 @@
+VOCABS['arabic'] = 'ءآأؤإئابةتثجحخدذرزسشصضطظعغـ‌‍‎‏فقكلمنهوىيًٌٍَُِّْٯٰپچژڜڢڤڥڧڨکگی' + '٠١٢٣٤٥٦٧٨٩'
```
Comment (fg-mindee) [inline arabic_vocab:1]: Would you mind removing this file please? :pray:

**Review by fg-mindee — COMMENTED** [review]
Thanks for the edits! Just a small modification and it looks like we're good to merge!

File: arabic_vocab (hunk 1 above)
Comment (mzeidhassan) [reply arabic_vocab]: I have deleted it. Hope it is all set now. Thanks @fg-mindee for your follow-up and support.

**Review by charlesmindee — CHANGES_REQUESTED** [review]
Thanks for the edit, you only need to split the last line in 2 lines to pass the flake8 test (code style enforcement, each line must be < 120) and we are good to merge (all the other tests are OK)!

**Comment by charlesmindee** [comment]
closes #490

File: doctr/datasets/vocabs.py [hunk 2]
```diff
@@ -15,6 +15,11 @@
     'punctuation': string.punctuation,
     'currency': '£€¥¢฿',
     'ancient_greek': 'αβγδεζηθικλμνξοπρστυφχψωΑΒΓΔΕΖΗΘΙΚΛΜΝΞΟΠΡΣΤΥΦΧΨΩ',
+    'arabic_letters': 'ءآأؤإئابةتثجحخدذرزسشصضطظعغـ‌‍‎‏فقكلمنهوىي',
+    'arabic_farsi_letters': 'ٯٰپچژڜڢڤڥڧڨکگی',
+    'hindi_numbers': '٠١٢٣٤٥٦٧٨٩',
+    'arabic_diacritics': 'ًٌٍَُِّْ',
+    'arabic_unique_punctuation': '؟؛«»—'
```
Comment (fg-mindee) [inline doctr/datasets/vocabs.py:22]: while you're at it, could you rename it `arabic_punctutation` please?

File: doctr/datasets/vocabs.py [hunk 3]
```diff
@@ -15,6 +15,11 @@
     'punctuation': string.punctuation,
     'currency': '£€¥¢฿',
     'ancient_greek': 'αβγδεζηθικλμνξοπρστυφχψωΑΒΓΔΕΖΗΘΙΚΛΜΝΞΟΠΡΣΤΥΦΧΨΩ',
+    'arabic_letters': 'ءآأؤإئابةتثجحخدذرزسشصضطظعغـ‌‍‎‏فقكلمنهوىي',
+    'arabic_farsi_letters': 'ٯٰپچژڜڢڤڥڧڨکگی',
```
Comment (fg-mindee) [inline doctr/datasets/vocabs.py:19]: Not entirely sure, but it looks like some characters are present in both?
Is that normal?

File: doctr/datasets/vocabs.py [hunk 4]
```diff
@@ -24,3 +29,4 @@
 VOCABS['portuguese'] = VOCABS['english'] + 'áàâãéêëíïóôõúüçÁÀÂÃÉËÍÏÓÔÕÚÜÇ' + '¡¿'
 VOCABS['spanish'] = VOCABS['english'] + 'áéíóúüñÁÉÍÓÚÜÑ' + '¡¿'
 VOCABS['german'] = VOCABS['english'] + 'äöüßÄÖÜẞ'
+VOCABS['arabic'] = VOCABS['arabic_letters'] + VOCABS['arabic_farsi_letters'] + VOCABS['hindi_numbers'] + VOCABS['digits'] + VOCABS['arabic_diacritics'] + VOCABS['arabic_unique_punctuation'] + VOCABS['punctuation']
```
Comment (fg-mindee) [inline doctr/datasets/vocabs.py:32]: One last thing: we have to make sure that there is no character duplicate (cf. my previous comment). Could you make sure that this is the case?

File: doctr/datasets/vocabs.py [hunk 5]
```diff
@@ -15,6 +15,11 @@
     'punctuation': string.punctuation,
     'currency': '£€¥¢฿',
     'ancient_greek': 'αβγδεζηθικλμνξοπρστυφχψωΑΒΓΔΕΖΗΘΙΚΛΜΝΞΟΠΡΣΤΥΦΧΨΩ',
+    'arabic_letters': 'ءآأؤإئابةتثجحخدذرزسشصضطظعغـ‌‍‎‏فقكلمنهوىي',
+    'arabic_farsi_letters': 'ٯٰپچژڜڢڤڥڧڨکگی',
+    'hindi_numbers': '٠١٢٣٤٥٦٧٨٩',
```
Comment (fg-mindee) [inline doctr/datasets/vocabs.py:20]: Could you rename it `hindi_digits` please? :pray:

File: doctr/datasets/vocabs.py (hunk 3 above)
Comment (fg-mindee) [reply doctr/datasets/vocabs.py:19]: Also, to make the name shorter, shouldnt "arabic_farsi_letters" be renamed to "persian_letters"?

File: doctr/datasets/vocabs.py (hunk 4 above)
Comment (fg-mindee) [reply doctr/datasets/vocabs.py:32]: To make flake8 content:
```python
VOCABS['arabic'] = (VOCABS['digits'] + VOCABS['hindi_digits'] + VOCABS['arabic_letters'] + VOCABS['persian_letters'] +
    VOCABS['arabic_diacritics'] + VOCABS['arabic_punctuation'] + VOCABS['punctuation'])
```

**Review by fg-mindee — CHANGES_REQUESTED** [review]
Thanks again, I added some suggestions to fix flake8 and improve naming :ok_hand:

File: doctr/datasets/vocabs.py (hunk 2 above)
Comment (charlesmindee) [reply doctr/datasets/vocabs.py:22]: `arabic_punctuation`

File: doctr/datasets/vocabs.py (hunk 3 above)
Comment (mzeidhassan) [reply doctr/datasets/vocabs.py:19]: @fg-mindee they may look similar, but they are different. In Arabic for example, we don't have 3 dots as in Farsi. Here are the Farsi characters:

> ٯ
ٰ
پ
چ
ژ
ڜ
ڢ
ڤ
ڥ
ڧ
ڨ
ک
گ
ی

For example, the last characters in that list, they are is a character that is very similar to Arabic, but their unicode points are different indeed.
Faris ی vs. Arabic ى

image

File: doctr/datasets/vocabs.py (hunk 5 above)
Comment (mzeidhassan) [reply doctr/datasets/vocabs.py]: Hi @fg-mindee, I believed I have made all requested changes. Can you please double check and let me know if there is anything I missed?

File: doctr/datasets/vocabs.py (hunk 3 above)
Comment (fg-mindee) [reply doctr/datasets/vocabs.py]: If they are indeed different, let's keep them but I'm already seeing issues if we can't tell the difference visually :sweat_smile:

File: doctr/datasets/vocabs.py (hunk 5 above)
Comment (fg-mindee) [reply doctr/datasets/vocabs.py]: Thanks! I'll run the CI checks to ensure we don't miss anything!

**Review by charlesmindee — APPROVED** [review]
Thanks for the updates!

**Review by fg-mindee — APPROVED** [review]
Thanks for edits, looks good to me!

**Pull request merged**